### PR TITLE
New stats page

### PR DESCRIPTION
**Pull request by cellocodingryan**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
None, this is a new page

### What is the new behavior?
New page for students to see current and past load on submitty for the current semester (fall or spring)

### Other information?
This pull request should not affect other parts of submitty

### CODE REVIEW

File: site/app/templates/Stats.twig [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            <a class="nav-bar normal-btn key_to_click timeframe" tabindex="0"  onclick="openStat(this, 'past_24','timeframe'), this.blur()">Past 24 Hours</a>
+            <a class="nav-bar normal-btn key_to_click timeframe" tabindex="0"  onclick="openStat(this, 'past_week','timeframe'), this.blur()">Past Week</a>
+            <a class="nav-bar normal-btn key_to_click timeframe" tabindex="0"  onclick="openStat(this, 'entire_semester','timeframe'), this.blur()">Entire Semester</a>
+        </div>
+    </div>
+    <div class =tab-bar-wrapper" style="margin-top: 15px;">
+        <a class="nav-bar normal-btn key_to_click typestab defaultOpen" tabindex="0"  onclick="openStat(this,'wait_time','typestab'), this.blur()">Wait Time</a>
+        <a class="nav-bar normal-btn key_to_click typestab" tabindex="0"  onclick="openStat(this,'grading_time','typestab'), this.blur()">Grading Time</a>
+    </div>
+    <div class =tab-bar-wrapper" style="margin-top: 15px;">
```
Comment (elihschiff) [inline site/app/templates/Stats.twig:20]:
```suggestion
    <div class="tab-bar-wrapper" style="margin-top: 15px;">
```

File: site/app/templates/Stats.twig [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+<div class="content" id="stat_page_content">
+    <div class="tab">
+        <div class =tab-bar-wrapper">
+            <a class="nav-bar normal-btn key_to_click timeframe defaultOpen" tabindex="0"  onclick="openStat(this, 'past_hour','timeframe'), this.blur()">Past Hour</a>
+            <a class="nav-bar normal-btn key_to_click timeframe" tabindex="0"  onclick="openStat(this, 'past_24','timeframe'), this.blur()">Past 24 Hours</a>
+            <a class="nav-bar normal-btn key_to_click timeframe" tabindex="0"  onclick="openStat(this, 'past_week','timeframe'), this.blur()">Past Week</a>
+            <a class="nav-bar normal-btn key_to_click timeframe" tabindex="0"  onclick="openStat(this, 'entire_semester','timeframe'), this.blur()">Entire Semester</a>
+        </div>
+    </div>
+    <div class =tab-bar-wrapper" style="margin-top: 15px;">
```
Comment (elihschiff) [inline site/app/templates/Stats.twig:16]:
```suggestion
    <div class="tab-bar-wrapper" style="margin-top: 15px;">
```

File: site/app/templates/Stats.twig [hunk 3]
```diff
@@ -0,0 +1,249 @@
+
+<style>
+    div.hidecontent {
+        display: none;
+    }
+</style>
+<div class="content" id="stat_page_content">
+    <div class="tab">
+        <div class =tab-bar-wrapper">
```
Comment (elihschiff) [inline site/app/templates/Stats.twig:9]:
```suggestion
        <div class="tab-bar-wrapper">
```

File: site/app/controllers/StatsController.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+use Symfony\Component\Routing\Annotation\Route;
+use app\models\OfficeHoursQueueModel;
+use app\libraries\routers\AccessControl;
+
+/**
+ * Class statscontroller
+ *
+ */
+
+class statscontroller extends AbstractController {
```
Comment (BenKelly99) [inline site/app/controllers/StatsController.php:18]: I believe we use UpperCamelCase for class names in Submitty, please use that

File: site/app/controllers/StatsController.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    public function __construct(Core $core) {
+        parent::__construct($core);
+    }
+
+    /**
+     * @Route("/{_semester}/{_course}/stats", methods={"GET"})
+     */
+
+    public function showall($something = null) {
```
Comment (BenKelly99) [inline site/app/controllers/StatsController.php:27]: Please use lowerCamelCase for method names in PHP
Additionally, what is the purpose of the $something = null parameter? It seems to have no effect

File: site/app/templates/Stats.twig [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+            <div class="wait_time typestab" id="entire_semester_wait"></div>
+            <div class="grading_time typestab" id="entire_semester_grade"></div>
+        </div>
+
+    </div>
+
+</div>
+<script>
+    var today = '{{ today }}';
```
Comment (BenKelly99) [inline site/app/templates/Stats.twig:77]: Please use let or const for variables in javascript

File: site/app/templates/Stats.twig [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+            },
+            yaxis:{
+                title: yaxis
+            },
+        });
+    }
+
+
+    function openStat(evt, statName, grouping) {
+        var tablinks = document.getElementsByClassName(grouping);
```
Comment (BenKelly99) [inline site/app/templates/Stats.twig:230]: Please use let for variable names in javascript

File: site/app/views/StatsView.php [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+            }
+            $next = floatval($next);
+            if($next != 0)
+                $array[] = array($date,$next);
+
+        }
+        return $array;
+    }
+
+    private function formatdate($date) {
```
Comment (BenKelly99) [inline site/app/views/StatsView.php:78]: Use lowerCamelCase for function names (applies to other functions in this file as well)

**Comment by elihschiff** [comment]
I am not sure how worried about performance we are but this page was very slow to load for me and I am worried it will be even slower to load when it is actually on Submitty and there is a lot of real data to parse.

Here is what chrome lighthouse says 
image

File: site/app/controllers/StatsController.php [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+ * Class StatsController
+ *
+ */
+class StatsController extends AbstractController {
+    public function __construct(Core $core) {
+        parent::__construct($core);
+    }
+
+    /**
+     * @Route("/{_semester}/{_course}/stats", methods={"GET"})
```
Comment (MasterOdin) [inline site/app/controllers/StatsController.php:24]:
```suggestion
     * @Route("/courses/{_semester}/{_course}/stats", methods={"GET"})
```

File: tests/e2e/accessibility_baseline.json [hunk 10]
```diff
@@ -81,5 +81,12 @@
         "The “font” element is obsolete. Use CSS instead.",
         "Element “div” not allowed as child of element “font” in this context. (Suppressing further errors from this subtree.)",
         "Too many messages."
+    ],
+    "/s20/sample/stats": [
```
Comment (MasterOdin) [inline tests/e2e/accessibility_baseline.json:85]:
```suggestion
    "/courses/s20/sample/stats": [
```

File: tests/e2e/test_accessibility.py [hunk 11]
```diff
@@ -52,6 +52,7 @@ def __init__(self, testname):
         '/{}/{}/late_table',
         '/{}/{}/theme',
         '/{}/{}/grades',
+        '/{}/{}/stats',
```
Comment (MasterOdin) [inline tests/e2e/test_accessibility.py:55]:
```suggestion
        '/courses/{}/{}/stats',
```

**Comment by bmcutler** [comment]
Closing this PR since the author is inactive and remaining work is non trivial.

**Pull request closed without merging**